evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -93,19 +95,21 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed X
         """
-        X_ww = infer_feature_types(X)
-        X_null_dropped = X_ww.ww.drop(self._all_null_cols)
-        if X_null_dropped.empty:
-            return X_null_dropped
+        X = infer_feature_types(X)
+        original_ltypes = X.ww.schema.logical_types
+        if len(self._all_null_cols) == X.shape[1]:
+            df = pd.DataFrame(index=X.index)
+            return _retain_custom_types_and_initalize_woodwork(original_ltypes, df)
```

Comment:
@freddyaboulton did our imputer behavior change here? I can't tell at first glance.

Reply:
No, just a small refactor to be able to do the drop with `inplace=True` and not create another dataframe in memory! The behavior of this code is tested by `test_imputer_empty_data` and `test_drop_all_columns`.